docs/proposals/service-catalog.md

```diff
@@ -0,0 +1,143 @@
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Abstract
+
+Kubernetes has Services but they are scoped per namespace, and (assuming eventual support for RBAC) you may not want everyone to be able to see and use all the services in your namespace.
+
+This proposal is about:
+
+- associating additional configuration/secret data with Services
+- adding a new Service Catalog resource into which users can publish services that others can easily find and consume
+- adding support for dynamic provisioning of resources from the catalog
+- making it easier to link Services (and their configuration data) to Pods
+
+# Use cases
+
+## Shared development database
+
+A development team is working on an application that uses a database. The IT department manages the database (i.e., it's lives off-cluster). All developers share the same credentials to access the database, but these credentials are managed by IT. Rather than having each developer create his or her own `Service` and `Secret` to connect to the database, IT creates a "db-app-xyz" `Service` and a "db-app-xyz" `Secret` in the "info-tech" namespace. IT also has lots of other `Service` resources in their namespace and they don't want to expose all of them to the development team. Therefore, they publish "db-app-xyz" to a service catalog. To use this service, a developer searches for it in the service catalog and adds it to their namespace.
+
+## Dynamic provisioning of cluster resources
+
+A user creates a "template" (e.g. if something similar to [OpenShift Templates](https://docs.openshift.org/latest/dev_guide/templates.html) exists) that makes it easy to create everything needed to spin up a new PostgreSQL database (customizable username/password, Service, Deployment, etc.). The user wants to share only this template in the service catalog so others can find it and use it, while keeping other templates in the namespace private.
+
+
+## Custom dynamic provisioning
+
+The IT department manages an off-cluster database. Each developer wanting to access the database is required to use a unique username and password to access the database. Additionally, each developer accesses a unique tablespace that no other team members can access. The IT department used to create database accounts and tablespaces by hand in response to individual developer requests.
+
+Moving forward, IT wants to automate the process. They create a web service that implements the "service broker" API (such as the one from [Cloud Foundry](http://docs.cloudfoundry.org/services/api.html#api-overview)). They add an entry to the service catalog for their database service, pointing at their web service. When a developer consumes the entry from the catalog, IT's web service is contacted, resulting in a new username, password, and tablespace.
```

Comment:
This might give the impression we can use an off-the-shelf CF Broker.  That won't actually work due to CF'isms in the broker implementations.  I agree we want the same workflow, though.